packages/scripts/utils/config.js

```diff
@@ -309,35 +312,49 @@ function getWebpackEntryPoints( buildType ) {
 					entryPoints[ entryName ] = entryFilepath;
 				}
 			}
-
-			if ( Object.keys( entryPoints ).length > 0 ) {
-				return entryPoints;
-			}
 		}
 
 		// Don't do any further processing if this is a module build.
 		// This only respects *module block.json fields.
 		if ( buildType === 'module' ) {
-			return {};
+			return entryPoints;
 		}
 
-		// 3. Checks whether a standard file name can be detected in the defined source directory,
-		//  and converts the discovered file to entry point.
-		const [ entryFile ] = glob( 'index.[jt]s?(x)', {
-			absolute: true,
-			cwd: fromProjectRoot( getWordPressSrcDirectory() ),
-		} );
+		const hasBlockJsonInRoot = blockMetadataFiles.includes(
+			join(
+				fromProjectRoot( getWordPressSrcDirectory() ),
+				'block.json'
+			).replace( /\\/g, '/' )
+		);
+		if ( ! hasBlockJsonInRoot ) {
+			// 3. Checks whether a standard JavaScript file name can be detected in the defined source directory,
+			//    and converts the discovered file to an entry point.
+			const [ entryJavaScriptFile ] = glob( 'index.?(m)[jt]s?(x)', {
```

Comment:
```suggestion
			const [ entryJavaScriptFile ] = glob( 'index.?(c)[jt]s?(x)', {
```

Reply:
I think we would also have to extend the support for CJS in the place where the loaders get configured:

https://github.com/WordPress/gutenberg/blob/7ce56960768287e833a70b76c2d6eb8788c69bbc/packages/scripts/config/webpack.config.js#L162